### DISCUSSION TITLE
Expanded relations formatting (hooks)

### DISCUSSION THREAD

**Discussion by k4io** [pocketbase/pocketbase#5905] [Q&A] [answered]
I have a setup with 5 collections:
licenses,products,users,user_roles,user_permissions

Each license has a single product relation
Each user can have multiple roles or specific permissions
Each role can have multiple permissions

A good example of my issue would be:
I have a license, the license has a field 'product' (relation, expanded). Depending on the users permissions (these are dynamic and user-created) I need to scramble/remove some of the data that is returned from the expanded field. I don't think I can achieve this due to the enriching being done at the end of the request.

I don't think I can apply a View rule here as the permissions are dynamic, some permissions will have JSON field 'conditions' which specifies which fields to remove or scramble (determined by notation, i.e: ['-cost', '~status']).

Is there a way around this or would I simply have to forget about using expand on the data I need to manipulate this way?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no option to apply modification after expand at the moment.

With the upcoming v0.23.0 (it is in the `develop` branch) this will be possible with the new `OnRecordEnrich` hook.
